pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -146,9 +146,15 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 	legacySymlink := legacyLogSymlink(containerID, containerMeta.Name, sandboxMeta.Name,
 		sandboxMeta.Namespace)
 	containerLog := filepath.Join(podSandboxConfig.LogDirectory, containerConfig.LogPath)
-	if err := m.osInterface.Symlink(containerLog, legacySymlink); err != nil {
-		glog.Errorf("Failed to create legacy symbolic link %q to container %q log %q: %v",
-			legacySymlink, containerID, containerLog, err)
+	// only create legacy symlink if containerLog path exists (or the error is not IsNotExist).
```

Comment:
what if the log file was created after the container started?

Reply:
The above is checked after the container is started and by now other two log  files (one new symlink in /var/log/pod and actual log file in container /var/lib/docker/container) are already created. So log file in /var/lib/docker/container is always exists. But the new symlink in /var/log/pod is only created if logging driver is json-file, but not if logging driver is journald (though not sure if there are other reasons if the new symlink in /var/log/pod wont be created.) Anyway, the end result is legacy symlink in /var/log/container will be dangling if the new symlink in /var/log/pod does not exist. This dangling, as I mentioned above, is removed by kube runtime gc at some point. But this PR fixes this to not create the dangling legacy symlink in the first place to address the issue here https://bugzilla.redhat.com/show_bug.cgi?id=1548987.

Reply:
ok, thanks for the explanation